src/v2/controllers/FormController.ts

```diff
@@ -255,13 +255,14 @@ export default Controller.extend({
 
     // Use full page redirection if necessary
     if (model.get('useRedirect')) {
-      // Clear when navigates away from SIW page, e.g. success, IdP Authenticator.
-      // Because SIW sort of finished its current /transaction/
-      sessionStorageHelper.removeStateHandle();
+      if (model.get('useRedirectButton')) {
+        // OKTA-635926: do not redirect without user gesture for ov enrollment on android
+        // We use a user gesture to complete the redirect in AutoRedirectView
+      } else {
+        // Clear when navigates away from SIW page, e.g. success, IdP Authenticator.
+        // Because SIW sort of finished its current /transaction/
+        sessionStorageHelper.removeStateHandle();
```

Comment:
Should we also add this in the button's click method?
https://github.com/okta/okta-signin-widget/pull/3782/files#diff-c992b801cae4c3d84efb925a6c392d7485753b37da4566e0992c90f07be564faR98-R100

Reply:
I think yes